### PR TITLE
Mixed coordinate system in curl

### PR DESCRIPTION
**Pull request by szymag**
This PR adds support for mixed coordinates in curl

### CODE REVIEW

File: sympy/vector/operators.py [hunk 1]
```diff
@@ -175,30 +175,22 @@ def curl(vect, coord_sys=None, doit=True):
...
-            return outvec.doit()
-        return outvec
+        if isinstance(vect, (Add, VectorAdd)):
+            return Add.fromiter(curl(i, doit=doit) for i in vect.args)
+        elif isinstance(vect, (Mul, VectorMul)):
+            vector = [i for i in vect.args if isinstance(i, (Vector, Cross, Gradient))][0]
+            scalar = Mul.fromiter(i for i in vect.args if not isinstance(i, (Vector, Cross, Gradient)))
+            res = Cross(gradient(scalar), vector) + Add.fromiter(scalar*i for i in curl(vector, doit=doit).args)
+            if doit:
+                try:
```
Comment (Upabjojr) [inline sympy/vector/operators.py]: no `try` here

Comment (szymag) [inline sympy/vector/operators.py]: OK

File: sympy/vector/operators.py [hunk 2]
```diff
@@ -175,30 +175,22 @@ def curl(vect, coord_sys=None, doit=True):
...
+                    return res.doit()
+                except:
+                    return res
+            return res
+        elif isinstance(vect, (Cross, Curl, Gradient)):
```
Comment (Upabjojr) [inline sympy/vector/operators.py]: Curl of the gradient is always zero: https://en.wikipedia.org/wiki/Vector_calculus_identities#Second_derivatives

Comment (szymag) [inline sympy/vector/operators.py]: Good point

Comment (Upabjojr) [inline sympy/vector/operators.py]: Same for the divergence of the curl.

File: sympy/vector/operators.py [hunk 3]
```diff
@@ -175,30 +175,22 @@ def curl(vect, coord_sys=None, doit=True):
...
-
-        if doit:
-            return outvec.doit()
-        return outvec
+        if isinstance(vect, (Add, VectorAdd)):
+            return Add.fromiter(curl(i, doit=doit) for i in vect.args)
+        elif isinstance(vect, (Mul, VectorMul)):
+            vector = [i for i in vect.args if isinstance(i, (Vector, Cross, Gradient))][0]
+            scalar = Mul.fromiter(i for i in vect.args if not isinstance(i, (Vector, Cross, Gradient)))
+            res = Cross(gradient(scalar), vector) + Add.fromiter(scalar*i for i in curl(vector, doit=doit).args)
```
Comment (Upabjojr) [inline sympy/vector/operators.py]: why is there `Add.fromiter`?

Comment (Upabjojr) [inline sympy/vector/operators.py]: I'd expect `Cross(gradient(scalar), vector) + scalar*curl(vector)`

Comment (szymag) [inline sympy/vector/operators.py]: Can't work. I tried that but there were some problem in constructor in `BasisDependentAdd` or `Mul`.
I need to multiply every component in curl separately

**Pull request merged**